File: Tutor Demos/Week 5/Python/team.py
import random

class Team:
    def __init__(self, name):
        self.__local_name = name.split()[0]
        self.__team_name = name.split()[1]
        self.__results = []
    
    def play(self):
        result = random.randint(1, 91) * (1 if random.choice([True, False]) else -1)
        self.__results.append(result)
        return result

    def best(self):
        #No checks
        highest = float('-inf')
        for result in self.__results:
            if result > highest:
                highest = result
        return highest

    def worst(self):
        #No checks
        smallest = float('inf')
        for result in self.__results:
            if result < smallest:
                smallest = result
        return smallest
    
    def average(self):
        total = 0
        for result in self.__results:
            total += result
        return total/len(self.__results)

    def __str__(self):
        return f"{self.__local_name} {self.__team_name}"

File: Tutor Demos/Week 5/Java/Team.java
import java.util.ArrayList;
import java.util.Random;

public class Team {
    private final String localName;
    private final String teamName;
    private final ArrayList<Integer> results;
    private final Random random;

    public Team(String name) {
        this.localName = name.split(" ")[0];
        this.teamName = name.split(" ")[1];
        this.results = new ArrayList<>();
        this.random = new Random();
    }

    public int play() {
        int result = random.nextInt(91) * (random.nextBoolean() ? 1 : -1);
        results.add(result);
        return result;
    }

    public int best() {
        //No checks
        int max = Integer.MIN_VALUE;
        for (Integer result : results) {
            if (result > max) {
                max = result;
            }
        }
        return max;
    }

    public int worst() {
        //No checks
        int min = Integer.MAX_VALUE;
        for (Integer result : results) {
            if (result < min) {
                min = result;
            }
        }
        return min;
    }

    public double average() {
        int sum = 0;
        for (Integer result : results) {
            sum += result;
        }
        return (double)sum/results.size();
    }

    @Override
    public String toString() {
        return localName + " " + teamName;
    }
}